tests/test_transpiler_unitary_decompositions.py

```diff
@@ -0,0 +1,206 @@
+import numpy as np
+import pytest
+from scipy.linalg import expm
+
+from qibo import gates, matrices
+from qibo.backends import NumpyBackend
+from qibo.models import Circuit
+from qibo.transpiler.unitary_decompositions import (
+    bell_basis,
+    calculate_h_vector,
+    calculate_psi,
+    calculate_single_qubit_unitaries,
+    cnot_decomposition,
+    cnot_decomposition_light,
+    magic_basis,
+    magic_decomposition,
+    to_bell_diagonal,
+    two_qubit_decomposition,
+)
+
+NREPS = 3  # number of repetitions to execute random tests
+ATOL = 1e-12
+
+
+def random_state(nqubits):
+    shape = 2**nqubits
+    psi = np.random.random(shape) + 1j * np.random.random(shape)
+    return psi / np.sqrt(np.sum(np.abs(psi) ** 2))
+
+
+def random_unitary(nqubits):
+    """Generates a random unitary matrix acting on nqubits."""
+    shape = 2 * (2**nqubits,)
+    m = np.random.random(shape) + 1j * np.random.random(shape)
+    return expm(1j * (m + np.conj(m.T)))
```

Comment:
`qibo` already has a built-in function for this in `quantum_info.random_ensembles`